react-web/src/components/templates/home/index.tsx

```diff
@@ -1,89 +1,127 @@
+import { ICurrencyProps } from 'components/molecules'
+import { WalletInput } from 'components/organisms/wallet-input'
+import { Currency, CurrencyLabel } from 'components/types/currency'
 
+import Eth from 'app/core/resources/eth.svg'
 import SwitchIcon from 'app/core/resources/switch.svg'
+import Weth from 'app/core/resources/weth.svg'
 
 import styles from './styles.module.scss'
 
 export interface IHomeTemplateProps {
   transactionTitle?: string
+  handleSubmit: (evt: FormEvent<HTMLFormElement>) => Promise<void>
 }
 
-interface IWalletInputProps {
-  currency: CurrencyLabel
-  isSender?: boolean
-}
+const HomeTemplate = ({ handleSubmit }: IHomeTemplateProps): JSX.Element => {
+  const [isButtonEnabled, setIsButtonEnabled] = useState(false)
+  const sendingRef = useRef<HTMLInputElement>(null)
+  const receivingRef = useRef<HTMLInputElement>(null)
+  const [toggleSwitchTitle, setToggleSwitchTitle] = useState(true)
+  const [inputSent, setInputSent] = useState('')
+  const [inputReceived, setInputReceived] = useState('')
+  const [currencyFrom, setCurrencyFrom] = useState(Currency.ETH)
+  const [currencyTo, setCurrencyTo] = useState(Currency.WETH)
 
-const WalletInput = ({
-  isSender,
-  currency,
-}: IWalletInputProps): JSX.Element => {
-  const settings = {
-    name: {
-      [CurrencyLabel.weth]: 'stellar wallet input',
-      [CurrencyLabel.eth]: 'ethereum wallet input',
+  const currencyPropsConverter: Record<Currency, ICurrencyProps> = {
+    [Currency.WETH]: {
+      initials: CurrencyLabel.weth,
+      label: Currency.WETH,
+      iconPath: Weth,
     },
-    label: isSender ? InputLabel.sending : InputLabel.receive,
-    button: {
-      [CurrencyLabel.weth]: 'Connect Stellar Wallet',
-      [CurrencyLabel.eth]: 'Connect Ethereum Wallet',
+    [Currency.ETH]: {
+      initials: CurrencyLabel.eth,
+      label: Currency.ETH,
+      iconPath: Eth,
     },
   }
 
-  return (
-    <div className={styles.inputContainer}>
-      <div className={styles.content}>
-        <Typography
-          className={styles.label}
-          variant={TypographyVariant.p}
-          text={isSender ? 'From:' : 'To:'}
-        />
-        <Button variant={ButtonVariant.secondary} size={ButtonSize.small}>
-          {settings.button[currency]}
-        </Button>
-      </div>
-      <LabeledInput
-        name={settings.name[currency]}
-        label={settings.label}
-        currency={currency}
-      />
-    </div>
-  )
-}
+  const onInputReceiveChange = (
+    evt: React.FormEvent<HTMLInputElement>
+  ): void => {
+    const input = evt.target as HTMLInputElement
+    setInputReceived(input.value)
+  }
+
+  const onInputSentChange = (evt: React.FormEvent<HTMLInputElement>): void => {
+    const input = evt.target as HTMLInputElement
+
+    setInputSent(input.value)
+  }
+
+  useEffect(() => {
+    if (inputSent > '0' && inputReceived > '0') {
+      setIsButtonEnabled(true)
+    } else {
+      setIsButtonEnabled(false)
```

Comment:
Here you can do like this too:
```suggestion
    setIsButtonEnabled(inputSent > '0' && inputReceived > '0')
```